14.serial_search/serial_search.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'date'
+
+def serial_search(input)
+  year = 1900 + (input[0] + input[4]).to_i
+  days = input[1..3].to_i
+  production = input[5..7].to_i
+
+  first_date = Date.new(year)
```

Comment:
```suggestion
  beginning_of_year = Date.new(year)
```

とか。
Railsにもそういう名前のメソッドがあるので。
https://api.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-beginning_of_year